### ISSUE TITLE
Allow `prefer-default-export` to report for files without default exports

### ISSUE THREAD

**Issue by imhoffd** [import-js/eslint-plugin-import#2600]
I noticed that the [prefer-default-export](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/prefer-default-export.md) rule only triggers when there is a single named export in a file. I would like for it to trigger for any file that doesn't have a default export: whether it has multiple named exports, a single named export, or no exports at all (but each of these could be configuration options for this rule).

The following patterns would be considered warnings:

```js
// bad1.js

// There is only a single module export and it's a named export.
export const foo = 'foo';
```

```js
// bad2.js

// There is more than one named export in the module.
export const foo = 'foo';
export const bar = 'bar';
```

The following patterns would not be considered warnings:

```js
// good1.js

// There is a single export and it's a default export.
const foo = 'foo';
export default foo;
```

```js
// good2.js

// There is a default export.
export const foo = 'foo';
const bar = 'bar';
export default bar;
```

**Comment by ljharb** [comment]
The default is built assuming that a file with multiple named exports is a bag of things - reducers, constants, etc. It certainly seems reasonable to add an option that would warn on any file missing a default export, though.

**Comment by imhoffd** [comment]
Thanks for that confirmation!

To elaborate on why this might be useful, in our codebase React components are default exports and their prop interfaces are named exports. It's easy enough to miss adding the `default` keyword to the component function, and this rule currently doesn't help with that.

In our particular case, our files are named the same as their default export (via [canonical/filename-match-exported](https://github.com/gajus/eslint-plugin-canonical#eslint-plugin-canonical-rules-filename-match-exported)). The idea is that files are self-contained modules revolving around a single, default declaration (the main thing) which may have zero to many named declarations (supplementary things that don't really warrant a file of their own, but are useful outside the module and as such are exported).
